lgc/builder/InOutBuilder.cpp

```diff
@@ -1655,17 +1655,20 @@ unsigned InOutBuilder::getBuiltInValidMask(BuiltInKind builtIn, bool isOutput) {
     HDGP = (1 << ShaderStageTessControl) | (1 << ShaderStageTessEval) | (1 << ShaderStageGeometry) |
            (1 << ShaderStageFragment),
     HG = (1 << ShaderStageTessControl) | (1 << ShaderStageGeometry),
-    MG = (1 << ShaderStageGeometry),
-    MVG = (1 << ShaderStageVertex) | (1 << ShaderStageGeometry),
-    MVDG = (1 << ShaderStageVertex) | (1 << ShaderStageTessEval) | (1 << ShaderStageGeometry),
-    MVHDG = (1 << ShaderStageVertex) | (1 << ShaderStageTessControl) | (1 << ShaderStageTessEval) |
-            (1 << ShaderStageGeometry),
+    M = (1 << ShaderStageMesh),
+    MG = (1 << ShaderStageMesh) | (1 << ShaderStageGeometry),
+    MVG = (1 << ShaderStageMesh) | (1 << ShaderStageVertex) | (1 << ShaderStageGeometry),
+    MVDG = (1 << ShaderStageMesh) | (1 << ShaderStageVertex) | (1 << ShaderStageTessEval) | (1 << ShaderStageGeometry),
```

Comment:
Instead of just changing the list, this could probably be easified by using a variadic template:

```MVG = GetBitmask<ShaderStageMesh, ShaderStageVertex, ShaderStageGeometry>();```

Reply:
Thank you. This is a good idea. I will try to make modifications.

Reply:
Maybe something like this works:

```
constexpr int GetBitMask() {
  return 0;
}

template<typename B, typename ...As>
constexpr int GetBitMask(B b, As... as) {
	return (1 << b) | GetBitMask(as...);
}
```

Would be easier to do in C++ 17, but well.